apis/v1alpha1/defaulting.go

```diff
@@ -243,7 +314,22 @@ func (in *GreptimeDBCluster) defaultMonitoringStandaloneSpec() *GreptimeDBStanda
 }
 
 func (in *GreptimeDBCluster) mergeFrontendTemplate() error {
-	if in.Spec.Frontend != nil {
+	if len(in.Spec.Frontends) != 0 {
+		for _, frontend := range in.Spec.Frontends {
```

Comment:
You can create a new function, for example, `doMergeFrontendTemplate()` to keep it clean.